File: VB/BackendMvcProject/Controllers/QueryBuilderController.vb
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc.Controllers


Namespace BackendMvcProject.Controllers
    ' This controller processes internal requests from the Query Builder that is a part of the Report Designer.
    Public Class QueryBuilderController
        Inherits QueryBuilderApiController

        Public Overrides Function Invoke() As ActionResult
            Dim result = MyBase.Invoke()
            ' Allow cross-domain requests.
            Response.AppendHeader("Access-Control-Allow-Origin", "*")
            Return result
        End Function
    End Class
End Namespace

File: CS/BackendMvcProject/Controllers/QueryBuilderController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc.Controllers;


namespace BackendMvcProject.Controllers {
    // This controller processes internal requests from the Query Builder that is a part of the Report Designer.
    public class QueryBuilderController : QueryBuilderApiController {
        public override ActionResult Invoke() {
            var result = base.Invoke();
            // Allow cross-domain requests.
            Response.AppendHeader("Access-Control-Allow-Origin", "*");
            return result;
        }
    }
}